packages/react/src/_VisuallyHidden.tsx

```diff
@@ -1,29 +1,25 @@
-import styled from 'styled-components'
+import classes from './_VisuallyHidden.module.css'
+import {clsx} from 'clsx'
+import {BoxWithFallback} from './internal/components/BoxWithFallback'
 import type {SxProp} from './sx'
-import sx from './sx'
 
 interface Props {
   isVisible?: boolean
+  as?: React.ElementType
 }
```

Comment:
I think if we want to have `as?` defined on `Props` then we would need get rid of the intersection below with `{ as?: As }` and make Props generic like:

```suggestion
interface Props<As extends React.ElementType> {
  isVisible?: boolean
  as?: As
}
```